packages/store-ui/src/molecules/Modal/ModalContent.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable jsx-a11y/no-noninteractive-tabindex */
+/*
+ * Disable the eslint rule to use sentinel html elements that aren't interactive.
+ */
+import type {
+  DetailedHTMLProps,
+  HTMLAttributes,
+  MouseEvent,
+  RefObject,
+} from 'react'
+import React, { useRef } from 'react'
+
+import useTrapFocus from './useTrapFocus'
+
+interface ModalContentPureProps
+  extends Omit<
+    DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
+    'ref'
+  > {
+  sentinelStartRef: RefObject<HTMLDivElement>
+  trapFocusRef: RefObject<HTMLDivElement>
+  sentinelEndRef: RefObject<HTMLDivElement>
+}
+
+const ModalContentPure = ({
+  trapFocusRef,
+  sentinelStartRef,
+  sentinelEndRef,
+  children,
+  ...props
+}: ModalContentPureProps) => {
+  return (
+    <>
+      <div tabIndex={0} data-testid="sentinelStart" ref={sentinelStartRef} />
+      {/*
+       * This next line is disabled due to the onClick prop.
+       * Even though div isn't clickable, the onClick is required to prevent the event bubbles
+       * until the overlay, which calls onDismiss inside the onClick handler, is clicked.
```

Comment:
Just rephrasing it, I was a little bit confused reading it before.

```suggestion
       * Even though a div isn't clickable, the onClick is required to prevent event bubbles
       * from reaching the Overlay, which calls onDismiss inside the onClick handler when it's clicked.
```